### PR TITLE
Feature/index handler

### PR DESCRIPTION
**Pull request by plally**
- add   `log_sql` config option and `ignore_auth` config option
- add `GET /suggestions` endpoint. 
  can be used with params `owner`, `sent` and `message_id` e.g. `GET /suggestions?sent=false`
- simplified storage interface, 
  now has `DeleteWhere` and `GetWhere` functions to handle all selects and deletes
- add created_at column that defaults to the current datetime

### CODE REVIEW

File: handlers_test.go [hunk 1]
```diff
@@ -23,7 +23,8 @@ func TestHandlers(t *testing.T) {
 		},
 	}
 
-	newSuggestion, _ := server.Create("23701337384550")
+	newSuggestion, _ := server.Create(&suggestions.Suggestion{Owner: "23701337384550"})
+	newSuggestionForDelete, _ := server.Create(&suggestions.Suggestion{Owner: "237012337384550"})
```
Comment (sarahsturgeon) [inline handlers_test.go]: Maybe assign this ID to a var?

File: suggestions/sqlite/sqlite.go [hunk 2]
```diff
@@ -95,14 +97,63 @@ func (store *SqliteSuggestionsStore) Update(suggestion *suggestions.Suggestion)
...
 	return err
 }
 
+func (store *SqliteSuggestionsStore) query(query string, args ...interface{}) (*sql.Rows, error) {
+	if store.LogQueries {
+		log.Infof("Sqlite query: \"%v\"  %v", query, args)
+	}
+	out, err := store.db.Query(query, args...)
+	if err != nil {
+		log.Info("Query errored: %v", err)
```
Comment (sarahsturgeon) [inline suggestions/sqlite/sqlite.go]: Is this the correct usage of `Info` and `Infof` ?
Looks like they're both formatting strings so I didn't know

**Review by sarahsturgeon — APPROVED** [review]
A couple of comments, nothing blocking.

Is this a drop-in upgrade or do I need to change something serverside?

File: suggestions/sqlite/sqlite.go (hunk 2 above)
Comment (plally) [reply suggestions/sqlite/sqlite.go]: yeah, that second one should be Infof as well

**Comment by plally** [comment]
> Is this a drop-in upgrade or do I need to change something serverside?

could you run 
```sql
ALTER TABLE cfc_suggestions ADD COLUMN created_at INTEGER DEFAULT 0
```
on the existing database

**Pull request merged**